### PR TITLE
refactor(all)Part 1 of changes to be complient with Phpstan level 2

### PR DESCRIPTION
**Pull request by wiewiurdp**
# Description

Changes to be complient with Phpstan level 2

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

- [ ] Unit Tests
- [ ] Behat Test

# Checklist:
- [x] I have read the contribution requirements and fulfill them.
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: module/exporter-shopware-6/src/Infrastructure/Client/Shopware6CategoryClient.php [hunk 1]
```diff
@@ -54,6 +55,10 @@ public function insert(
         $action = new PostCategoryAction($shopwareCategory, true);
 
         $newShopwareCategory = $this->connector->execute($channel, $action);
+
+        if (!$newShopwareCategory instanceof Shopware6Category) {
+            throw new UnexpectedTypeException($newShopwareCategory, Shopware6Category::class);
```
Comment (piotrkreft) [inline module/exporter-shopware-6/src/Infrastructure/Client/Shopware6CategoryClient.php:60]: You shouldn't throw validator exception

File: module/product/src/Domain/Query/ProductQueryInterface.php [hunk 2]
```diff
@@ -91,6 +92,12 @@ public function findProductIdByType(string $type): array;
 
     public function getCount(): int;
 
+    /**
+     *
+     * @return array
```
Comment (piotrkreft) [inline module/product/src/Domain/Query/ProductQueryInterface.php:97]: Is it possible to declare a traversable type?

File: module/exporter-shopware-6/src/Infrastructure/Client/Shopware6CategoryClient.php (hunk 1 above)
Comment (piotrkreft) [reply module/exporter-shopware-6/src/Infrastructure/Client/Shopware6CategoryClient.php:60]: So it goes for the rest non-validator places

**Review by piotrkreft — CHANGES_REQUESTED** [review]

File: module/attribute/src/Domain/Provider/AttributeParametersProvider.php [hunk 3]
```diff
@@ -35,6 +36,9 @@ public function provide(AbstractAttribute $attribute): array
         if (isset($parameters['unit'])) {
             $unit = $this->unitRepository->load(new UnitId($parameters['unit']));
             if ($unit) {
```
Comment (BastekBielawski) [inline module/attribute/src/Domain/Provider/AttributeParametersProvider.php:38]: These make no sense for me. Use `if (!$unit instanceof Unit)`

File: module/attribute/src/Infrastructure/Mapper/Strategy/DateAttributeMapperStrategy.php [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+class DateAttributeMapperStrategy implements AttributeMapperStrategyInterface
+{
+    public function supported(AttributeType $type): bool
+    {
+        return $type->getValue() === DateAttribute::TYPE;
+    }
+
+    public function map(array $values): ValueInterface
+    {
+        Assert::allRegex(array_keys($values), '/^[a-z]{2}_[A-Z]{2}$/');
```
Comment (BastekBielawski) [inline module/attribute/src/Infrastructure/Mapper/Strategy/DateAttributeMapperStrategy.php:27]: We check all keys and we don't use it? Why we check keys if they those not matter?

File: module/attribute/src/Infrastructure/Mapper/Strategy/DateAttributeMapperStrategy.php [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+        foreach ($values as $value) {
+            if (null !== $value) {
```
Comment (BastekBielawski) [inline module/attribute/src/Infrastructure/Mapper/Strategy/DateAttributeMapperStrategy.php:29]: Better solution will be `array_filter` first

File: module/attribute/src/Infrastructure/Mapper/Strategy/FileAttributeMapperStrategy.php [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+    {
+        return $type->getValue() === FileAttribute::TYPE;
+    }
+
+    public function map(array $values): ValueInterface
+    {
+        Assert::allRegex(array_keys($values), '/^[a-z]{2}_[A-Z]{2}$/');
+        Assert::allIsArray($values);
+        foreach ($values as $language => $value) {
+            if (is_array($value) && !empty($value)) {
```
Comment (BastekBielawski) [inline module/attribute/src/Infrastructure/Mapper/Strategy/FileAttributeMapperStrategy.php:29]: I don't get it. We check `Assert::allIsArray`, why we doing it second time in condition?

File: module/attribute/src/Infrastructure/Mapper/Strategy/GalleryAttributeMapperStrategy.php [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+    {
+        return $type->getValue() === GalleryAttribute::TYPE;
+    }
+
+    public function map(array $values): ValueInterface
+    {
+        Assert::allRegex(array_keys($values), '/^[a-z]{2}_[A-Z]{2}$/');
+        Assert::allIsArray($values);
+        foreach ($values as $language => $value) {
+            if (is_array($value) && !empty($value)) {
```
Comment (BastekBielawski) [inline module/attribute/src/Infrastructure/Mapper/Strategy/GalleryAttributeMapperStrategy.php:29]: I don't get it. We check Assert::allIsArray, why we doing it second time in condition?

File: module/attribute/src/Infrastructure/Mapper/Strategy/ImageAttributeMapperStrategy.php [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+    public function supported(AttributeType $type): bool
+    {
+        return $type->getValue() === ImageAttribute::TYPE;
+    }
+
+    public function map(array $values): ValueInterface
+    {
+        Assert::allRegex(array_keys($values), '/^[a-z]{2}_[A-Z]{2}$/');
+        foreach ($values as $value) {
+            if (null !== $value) {
```
Comment (BastekBielawski) [inline module/attribute/src/Infrastructure/Mapper/Strategy/ImageAttributeMapperStrategy.php:29]: `array_filter` will be better solution

**Review by BastekBielawski — CHANGES_REQUESTED** [review]

File: module/account/src/Infrastructure/Handler/Role/UpdateRoleCommandHandler.php [hunk 9]
```diff
@@ -29,7 +28,16 @@ public function __construct(RoleRepositoryInterface $repository)
...
     {
         $role = $this->repository->load($command->getId());
-        Assert::isInstanceOf($role, Role::class, sprintf('Can\'t find Role with id %s', $command->getId()));
+
+        if (!$role instanceof Role) {
+            throw new \LogicException(
+                sprintf(
+                    'Expected an instance of %s. %s received.',
+                    Role::class,
+                    get_debug_type($role)
```
Comment (wfajczyk) [inline module/account/src/Infrastructure/Handler/Role/UpdateRoleCommandHandler.php:37]: This function has been added to PHP 8. This project currently using 7.4
https://wiki.php.net/rfc/get_debug_type

**Review by wfajczyk — CHANGES_REQUESTED** [review]

File: module/account/src/Infrastructure/Handler/Role/UpdateRoleCommandHandler.php (hunk 9 above)
Comment (wiewiurdp) [reply module/account/src/Infrastructure/Handler/Role/UpdateRoleCommandHandler.php:37]: @piotrkreft czy ja dobrze rozumiem, że jeżeli widze w symfony.locku, że jest instalowany symfony/polyfill-php80 to on dostarcza tą funkcje?

Comment (piotrkreft) [reply module/account/src/Infrastructure/Handler/Role/UpdateRoleCommandHandler.php:37]: Yup, it's part of the installed polyfill. Safe to use and FC

**Review by piotrkreft — APPROVED** [review]

**Review by wfajczyk — APPROVED** [review]

**Review by piotrkreft — APPROVED** [review]

**Review by BastekBielawski — APPROVED** [review]

**Pull request merged**